Jenkinsfile

```diff
@@ -193,6 +193,33 @@ pipeline {
                 }
             }
         }
+        stage("Make-doxygen") {
+            agent any
+            steps {
+              script {
+                if (isBranch('master')) {
+                  withCredentials([usernamePassword(credentialsId: 'a630aebc-6861-4e69-b497-fd7f496ec46b',
+                      usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_PASSWORD')]) {
+                      sh """#!/bin/bash
+                          set -x
+                          make doxygen
+                          git config --global user.email "mc.stanislaw@gmail.com"
+                          git config --global user.name "Stan Jenkins"
+                          git checkout --detach
+                          git branch -D gh-pages
+                          git checkout --orphan gh-pages
+                          git rm --cached stan test lib make
+                          git add -f doc
+                          git commit -m "auto generated docs from Jenkins"
+                          git push https://${GIT_USERNAME}:${GIT_PASSWORD}@github.com/${fork()}/math.git +gh-pages:gh-pages
+                          git checkout -f master
+                          exit 1
+                          """
```

Comment:
1. I added a command here to remove `stan`, `test`, `lib` and `make` folders so that it would push to gh-pages faster, is that okay?

2. Do I need that forced checkout back to master at the end?